packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -10,32 +10,59 @@
  * governing permissions and limitations under the License.
  */
 
+import {getScrollParent} from './';
+
+let isScrollPrevented = false;
+
+export function setScrollPrevented(value: boolean) {
+  isScrollPrevented = value;
+}
+
+interface ScrollOpts {
+  scrollView: HTMLElement
+}
+
 /**
  * Scrolls `scrollView` so that `element` is visible.
  * Similar to `element.scrollIntoView({block: 'nearest'})` (not supported in Edge),
  * but doesn't affect parents above `scrollView`.
```

Comment:
Mm, yeah I can see how it would be confusing now with `opts: HTMLElement | ScrollOpts` as the signature. I kinda want to bring back the overloads that were removed [here](https://github.com/adobe/react-spectrum/pull/3899/commits/371eede4b6222aa2d0de5c10fa0fb6655bebc48c) since changing this description to be "Scrolls opts.scrollView so that element is visible" feels a bit unclear, especially since `opts` is actually the `element` in the old signature that this still supports.

@snowystinger how do you feel about the above?

Reply:
```
/**
 * Scrolls `scrollView` in ScrollOptions so that `element` is visible.
 * Similar to `element.scrollIntoView({block: 'nearest'})` (not supported in Edge),
 * but doesn't affect parents above `scrollView`.
 * `scollView can be passed directly in place of the ScrollOptions.`
```
and whenever we add any options to the ScrollOpts, we can add to the last sentence. ', as a shorthand for *whatever default options we want to specify*'

Reply:
does that feel better?

Reply:
Well, thats the thing, `scrollView` doesn't actually substitute as `opts` if passed directly since the old signature was the following:
`scrollIntoView(scrollView: HTMLElement, element: HTMLElement)` 
versus the new signature
`scrollIntoView(element: HTMLElement, opts: HTMLElement | ScrollOpts)`

This means `element` is actually mapped to `scrollView` and `opts` is mapped to `element` if the user is providing a `HTMLElement` to opts (aka they are using the old signature). This feels pretty confusing and hard to specify clearly in a single description :/

Reply:
ahhhhh, sorry, i got mixed up then. please keep the override syntax